### ISSUE TITLE
View Registry render() syntax

### ISSUE THREAD

**Issue by gbdrummer** [ngn-jet/jet#3]
Currently the View Registry render() function takes a bunch of arguments in a couple possible configurations, and I often find myself forgetting what the order is and end up looking it up in the docs. 

In addition, the callback functionality is currently broken, as it does not return a reference to the new element created. I have also found the callback syntax to be a bit clunky in practice, especially when combined with all the other arguments. It occurred to me that a promise might actually provide a very nice way to improve the usability of the render() function.

I suggest this syntax:
```js
Registry.render({
  template: 'template-name',
  variables: {
    varName1: 'value 1',
    varName2: 'value 2'
  },
  target: targetElement
}).then(ref => {
  ...do stuff like add eventlisteners...
}).catch(err => {
  if (err) {
    throw err
  }
})
```

The then() callback would pass a NGN.DOM.Reference to the newly created element as an argument. 

I've also found in practice that guaranteeChild() may not be necessary, at least not in all cases. a simple setTimeout with 0 for the duration consistently works. I'd have to look into it a bit deeper to see why this is the case, but so far it always works.

**Comment by coreybutler** [comment]
The promise syntax is certainly possible, though it forces use of a legacy edition on IE11 (which I'm fine with). It wouldn't be the first request for promises either. 

My concern is everything else is event based. The events are often used to change state elsewhere in the view registry. That said, I see no reason not to return a promise from the render method, since I think it's just a void function right now.The key to usability would be making sure people know that both promises and events are possible, mostly to prevent conflicts if they're using both.

**Comment by gbdrummer** [comment]
I agree- I'm finding in practice that I am mostly using this method in conjunction with NGN.Tasks (or NGN.Queue now), generating multiple elements within a loop- That makes the event syntax a little odd to work with since I'd have to apply a new listener on each iteration to fire the task completion callback.

It would still be nice to have the event syntax for one-off renders though.

**Comment by coreybutler** [comment]
Would it make sense to bake the NGN.Queue into the view registry (via pointer, not duplicate code)?

**Comment by gbdrummer** [comment]
I could see us possibly coming up with a more streamlined way of rendering DOM elements within Registries. I'd have to think about what would be the best way to do that- would like to hear your thoughts.

**Comment by gbdrummer** [comment]
I'd like to add an additional possibility for a feature of the `render()` function:

When an template is rendered into a View Registry, the Registry should automatically store reference to it which can be accessed by a custom id. This way, when working with the new element after it has been rendered it will not be necessary to run additional querySelector calls. 

One of the things that is nice about working with React from a workflow perspective is that since you can just add listeners directly to the JSX, you don't need to do the work of querying for elements and manually adding event listeners. I think that anything we can do to make this process as painless as possible in JET will help it rival other frameworks, and improve our own workflow.
